### PR TITLE
Migrate to es5

### PR DESCRIPTION
**Pull request by shvaikalesh**

### CODE REVIEW

**Comment by paulmillr** [comment]
Can you squish all commits into one?

File: src/chaplin/controllers/controller.coffee [hunk 1]
```diff
@@ -63,9 +63,10 @@ module.exports = class Controller
     return if @disposed
 
     # Dispose and delete all members which are disposable.
-    for own prop, obj of this when obj and typeof obj.dispose is 'function'
-      obj.dispose()
-      delete this[prop]
+    Object.keys(@).forEach (key) =>
```
Comment (paulmillr) [inline src/chaplin/controllers/controller.coffee]: `this`, don't use single `@`, this is described in code style

File: src/chaplin/controllers/controller.coffee [hunk 2]
```diff
@@ -63,9 +63,10 @@ module.exports = class Controller
...
+      if (object = @[key]) and typeof object.dispose is 'function'
```
Comment (paulmillr) [inline src/chaplin/controllers/controller.coffee]: don't assign in `if`s too

File: src/chaplin/lib/utils.coffee [hunk 3]
```diff
@@ -1,34 +1,9 @@
...
 # Utilities
 # ---------
 
 utils =
-  # Object Helpers
-  # --------------
-
-  # Prototypal delegation. Create an object which delegates
-  # to another object.
-  beget: do ->
```
Comment (paulmillr) [inline src/chaplin/lib/utils.coffee]: Let's keep this as an alias for `Object.create`.

File: src/chaplin/lib/utils.coffee [hunk 4]
```diff
@@ -1,34 +1,9 @@
...
-    if typeof Object.create is 'function'
-      Object.create
-    else
-      ctor = ->
-      (obj) ->
-        ctor.prototype = obj
-        new ctor
-
-  indexOf: do ->
```
Comment (paulmillr) [inline src/chaplin/lib/utils.coffee]: Let's keep this for existing apps.

File: src/chaplin/lib/utils.coffee [hunk 5]
```diff
@@ -1,34 +1,9 @@
...
-    if Array::indexOf
-      (list, index) -> list.indexOf index
-    else if _.indexOf
-      _.indexOf
-
-  isArray: Array.isArray or _.isArray
```
Comment (paulmillr) [inline src/chaplin/lib/utils.coffee]: Same here

File: src/chaplin/lib/utils.coffee (hunk 3 above)
Comment (paulmillr) [inline src/chaplin/lib/utils.coffee]: Nevermind, saw it below

File: src/chaplin/mediator.coffee [hunk 6]
```diff
@@ -47,13 +46,11 @@ mediator.setHandler = (name, method, instance) ->
   handlers[name] = {instance, method}
 
 # Retrieves a handler function and executes it.
-mediator.execute = (nameOrObj, args...) ->
-  silent = false
-  if typeof nameOrObj is 'object'
-    silent = nameOrObj.silent
-    name = nameOrObj.name
+mediator.execute = (options, args...) ->
+  if options is Object(options)
```
Comment (paulmillr) [inline src/chaplin/mediator.coffee]: This would match functions. It wasn't matching functions before

File: src/chaplin/lib/utils.coffee (hunk 3 above)
Comment (shvaikalesh) [inline src/chaplin/lib/utils.coffee]: Yop, we cannot remove it cause it is public API.

File: src/chaplin/controllers/controller.coffee (hunk 1 above)
Comment (shvaikalesh) [inline src/chaplin/controllers/controller.coffee]: Okay, will fix that and add linter rule.

File: src/chaplin/mediator.coffee (hunk 6 above)
Comment (shvaikalesh) [inline src/chaplin/mediator.coffee]: `x === Object(x)` is used to test method parameters if they are not primitives and the change does not introduce any backwards-incompatible changes. I am trying to avoid throws on passed `null`.

Functions are just objects that can be called. If we are going to treat them as objects (pass multiple parameters, use as hashes) this check is ok. If we are going to call some value, then it is `typeof is 'function'`

**Comment by shvaikalesh** [comment]
Squashed the commits, brought back `typeof` stuff.

File: src/chaplin/controllers/controller.coffee [hunk 7]
```diff
@@ -63,9 +63,11 @@ module.exports = class Controller
     return if @disposed
 
     # Dispose and delete all members which are disposable.
-    for own prop, obj of this when obj and typeof obj.dispose is 'function'
-      obj.dispose()
-      delete this[prop]
+    Object.keys(this).forEach (key) =>
+      object = this[key]
+      if object and typeof object.dispose is 'function'
+        object.dispose()
```
Comment (vendethiel) [inline src/chaplin/controllers/controller.coffee]: Why this change? Doesn't the original read better?

Comment (shvaikalesh) [inline src/chaplin/controllers/controller.coffee]: I don't like the generated code.
